test/integration/job/job_test.go

```diff
@@ -2985,3 +3004,32 @@ func updateJob(ctx context.Context, jobClient typedv1.JobInterface, jobName stri
 	})
 	return job, err
 }
+
+func addFinalizerAndDeletePods(ctx context.Context, t *testing.T, clientSet clientset.Interface, namespace string) {
+	t.Helper()
+	pods, errList := clientSet.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{})
+	if errList != nil {
+		t.Fatalf("Failed to list pods: %v", errList)
+	}
+	updatePod(t, clientSet, pods.Items, func(pod *v1.Pod) {
+		pod.Finalizers = append(pod.Finalizers, "fake.example.com/blockDeletion")
+	})
+	err := clientSet.CoreV1().Pods(namespace).DeleteCollection(ctx,
+		metav1.DeleteOptions{},
+		metav1.ListOptions{
+			Limit: 1000,
+		})
+	if err != nil {
+		t.Fatalf("Failed to cleanup Pods: %v", err)
+	}
+
+	podsDelete, errList2 := clientSet.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{})
+	if errList2 != nil {
+		t.Fatalf("Failed to list pods: %v", errList2)
+	}
+	for _, val := range podsDelete.Items {
+		if val.DeletionTimestamp == nil {
+			t.Fatalf("Deletion not registered.")
+		}
+	}
```

Comment:
This seems unnecessary. This is tested already for the apiserver

Reply:
good catch